### PR TITLE
Add CONTRIBUTING.md: Guidelines for Contributing to HackTheMountains Discord Bot

### PR DESCRIPTION
**Pull request by imkrishnasarathi**
This pull request adds a CONTRIBUTING.md file to the project, providing guidelines and instructions for contributors. Having clear contribution guidelines will help streamline the contribution process and improve collaboration among community members. 
PS - It took me 2 hours since I'm a beginner in writing a documentation!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    - [3. Make Changes](#3-make-changes)
+    - [4. Review and Collaboration](#4-review-and-collaboration)
+    - [5. Acknowledgment](#5-acknowledgment)
+  - [Pull Requests](#pull-requests)
+- [Styleguides](#styleguides)
+  - [Git Commit Messages](#git-commit-messages)
+  - [JavaScript or TypeScript Styleguide](#javascript-or-typescript-styleguide)
+  - [Documentation Styleguide](#documentation-styleguide)
+
+
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:35]: Not keeping this because github autogenerates one

```suggestion
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+## I have a Question
+
+If you want to ask a question, we assume that you have read the available [Documentation](https://github.com/xkrishguptaa/bot-dc-htm/blob/main/README.md).
+
+Before you ask a question, it is best to search for existing [Issues](https://github.com/xkrishguptaa/bot-dc-htm/issues) that might help you. In case you found a suitable issue and still need clarification, you can write your question in the issue. It is also advisable to search the internet for answers first.
+
+If you then still feel the need to ask a question and need clarification, we recommend the following:
+
+- Open an [Issue](https://github.com/xkrishguptaa/bot-dc-htm/issues/new).
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:48]:
```suggestion
- Join us on [Discord](https://bit.ly/HackTheMountains) and ask it!
- Open an [Issue](https://github.com/xkrishguptaa/bot-dc-htm/issues/new).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+- **Explain the behavior you expected to see instead and provide the reasoning behind your expectation.**
+- **Enhance your report with screenshots or animated GIFs** that visually demonstrate the problem as you follow the described steps.
+- **If the issue wasn't triggered by a specific action, describe what you were doing before the problem occurred and provide any relevant additional information.**
+
+Once it's done:
+
+- The project team will label the issue accordingly.
+- A team member will try to reproduce the issue with your provided steps. If there are no reproduction steps or no obvious way to reproduce the issue, the team will ask you for those steps and mark the issue as `needs-repro`. Bugs with the `needs-repro` tag will not be addressed until they are reproduced.
+- If the team is able to reproduce the issue, it will be marked `needs-fix`, as well as possibly other tags (such as `critical`), and the issue will be left to be implemented by someone.
+
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:86]:
```suggestion
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+Explain the method used, whether it's via mouse clicks, keyboard shortcuts, or other actions, and specify which ones.
+- **Offer specific examples** like links to relevant files or projects, or code snippets that illustrate the issue.
+- **Detail what behavior you observed after following the steps** and clearly identify the problem within that behavior.
+- **Explain the behavior you expected to see instead and provide the reasoning behind your expectation.**
+- **Enhance your report with screenshots or animated GIFs** that visually demonstrate the problem as you follow the described steps.
+- **If the issue wasn't triggered by a specific action, describe what you were doing before the problem occurred and provide any relevant additional information.**
+
+Once it's done:
+
+- The project team will label the issue accordingly.
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:83]:
```suggestion
Once it's done we will review and triage the issue accordingly.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+- Use a **clear and descriptive title** for the issue to identify the suggestion.
+- Provide a **step-by-step description of the suggested enhancement** in as many details as possible.
+- **Provide specific examples to demonstrate the steps**. You may include copy/pasteable snippets which you use in those examples.
+- You may want to **include screenshots and animated GIFs** which help you demonstrate the steps or point out the part which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux. 
+- **Explain why this enhancement would be useful** to most of the users. You may also want to point out the other projects that solved it better and which could serve as inspiration.
+
+### Your First Code Contribution 🎊
+
+Unsure where to begin contributing? You can start by looking through these `beginner` and `help-wanted` or `good first issue` labeled issues:
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:110]:
```suggestion
Unsure where to begin contributing? You can start by looking through these `help-wanted` or `good first issue` labeled issues:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+- Beginner Issues are very simple issues and require very less work.
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:112]:
```suggestion
```

**Review by xkrishguptaa — CHANGES_REQUESTED** [review]
Hi! Dropped some suggestions, also can you please remove the `styleguide` section as we are not having any specific styles right now? 

Thank you for your contribution

**Comment by imkrishnasarathi** [comment]
Take a look !

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+- Maintain quality
+- Fix problems that are important to users
+- Engage the community in working toward making this project better.
+- Enable a sustainable system for the maintainers to review contributions.
+
+To submit a better Pull Request, follow these:
+
+1. If applicable, use appropriate templates for the Pull Requests.
+2. Follow the [styleguides](#styleguides)
```
Comment (xkrishguptaa) [inline CONTRIBUTING.md:136]:
```suggestion
```

**Review by xkrishguptaa — APPROVED** [review]
Looks good

**Comment by xkrishguptaa** [comment]
Congratulations on your PR getting merged :clap: We are gifting HTM Coins (in-server currency) to those who are contributing to our repo.

You can join the [HTM Discord](https://bit.ly/HackTheMountains) to claim the coins! :D Ping me `@xkrishguptaa` on discord to know more

**Pull request merged**